pkg/services/mysqldb/provision_test.go

```diff
@@ -0,0 +1,66 @@
+package mysqldb
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateNoFirewallConfig(t *testing.T) {
+
+	sm := &serviceManager{}
+
+	pp := &ProvisioningParameters{}
+
+	error := sm.ValidateProvisioningParameters(pp)
+	assert.Nil(t, error)
+}
+
+func TestValidateGoodFirewallConfig(t *testing.T) {
+
+	sm := &serviceManager{}
+
+	pp := &ProvisioningParameters{
+		FirewallIPStart: "192.168.86.1",
+		FirewallIPEnd:   "192.168.86.100",
+	}
+
+	error := sm.ValidateProvisioningParameters(pp)
+	assert.Nil(t, error)
+}
+
+func TestValidateMissingEndFirewallConfig(t *testing.T) {
+
+	sm := &serviceManager{}
+
+	pp := &ProvisioningParameters{
+		FirewallIPStart: "192.168.86.1",
+	}
+
+	error := sm.ValidateProvisioningParameters(pp)
+	assert.NotNil(t, error)
```

Comment:
Can we assert that the error is a `service.Validation` error and that the `Field` attribute of that error is the one we expected? Same comment applies for test cases below.